controllers/clusterresources/opensearchuser_controller.go

```diff
@@ -205,19 +205,31 @@ func (r *OpenSearchUserReconciler) createUser(
 		return err
 	}
 
-	err = r.API.CreateUser(user.ToInstaAPI(username, password), clusterID, models.OpenSearchAppKind)
+	exists, err := CheckIfUserExists(username, clusterID, models.OpenSearchAppKind, r.API)
 	if err != nil {
-		logger.Error(err, "Cannot create OpenSearch user on Instaclustr",
-			"username", username,
-			"cluster ID", clusterID,
-		)
+		logger.Error(err, "Cannot check if user exists ")
```

Comment:
Have you tested a case when a cluster was deleted? What error code do we get?

Reply:
I have tested it, we getting 404 not found

Reply:
I have tested it, we getting 404 not found

Reply:
The error is returned at the endpoint when we fetch all users in a cluster. 
For now, let's just handle it as a regular error.